app/admin/views.py

```diff
@@ -183,3 +185,70 @@ def delete_user(user_id):
         db.session.commit()
         flash('Successfully deleted user %s.' % user.full_name(), 'success')
     return redirect(url_for('admin.registered_users'))
+
+
+@admin.route('/agencies')
+@login_required
+@admin_required
+def all_agencies():
+    """View all agencies."""
+    agencies = Agency.query.all()
+    return render_template('admin/all_agencies.html', agencies=agencies)
+
+
+@admin.route('/agency/<int:agency_id>')
+@admin.route('/agency/<int:agency_id>/info')
+@login_required
+@admin_required
+def agency_info(agency_id):
+    """View an agency's information."""
+    agency = Agency.query.filter_by(id=agency_id).first()
+    if agency is None:
+        abort(404)
+    return render_template('admin/manage_agency.html', agency=agency)
+
+
+@admin.route('/agency/<int:agency_id>/change-official-status',
+             methods=['GET', 'POST'])
+@login_required
+@admin_required
+def change_agency_official_status(agency_id):
+    """Make an agency official or unofficial."""
+    agency = Agency.query.filter_by(id=agency_id).first()
+    if agency is None:
+        abort(404)
+    form = ChangeAgencyOfficialStatusForm()
+    if form.validate_on_submit():
+        agency.is_official = form.is_official.data == 'y'
```

Comment:
hahaha I knew someone would ask

_cries_

http://stackoverflow.com/questions/33429510/wtforms-selectfield-not-properly-coercing-for-booleans/33430086?noredirect=1#comment54649785_33430086